torch/csrc/autograd/engine.cpp

```diff
@@ -582,13 +582,19 @@ void GraphTask::exec_post_processing() {
     //  2. The callback's results can safely be used on (user-facing) caller_current_streams
     //     after backward().
     c10::MultiStreamGuard g(caller_current_streams_filtered);
-    // WARNING: Don't use a range-for loop here because more callbacks may be
-    // added in between callback calls, so iterators may become invalidated.
-    // NOLINTNEXTLINE(modernize-loop-convert)
-    for (size_t i = 0; i < final_callbacks_.size(); ++i) {
-      cb_lock.unlock();
-      final_callbacks_[i]();
-      cb_lock.lock();
+    {
```

Comment:
There is already a code block just above the comment above, no need for a new one here.